source/slang/slang-emit-c-like.cpp

```diff
@@ -3704,6 +3714,234 @@ void CLikeSourceEmitter::emitFuncDecorationsImpl(IRFunc* func)
     }
 }
 
+bool CLikeSourceEmitter::tryGetIntInfo(IRType* elementType, bool &isSigned, int &bitWidth)
+{
+    Slang::IROp type = elementType->getOp();
+    if (!(type >= kIROp_Int8Type && type <= kIROp_UInt64Type)) return false;
+    isSigned = (type >= kIROp_Int8Type && type <= kIROp_Int64Type);
+
+    Slang::IROp stype = (isSigned) ? type : Slang::IROp(type - 4);
+    bitWidth = 8 << (stype - kIROp_Int8Type);
+    return true;
+}
+
+void CLikeSourceEmitter::emitVecNOrScalar(IRVectorType* vectorType, std::function<void()> emitComponentLogic)
```

Comment:
naming suggestion: `maybeEmitMakeVectorFromScalar`. Instead of doing call-back style with std::function (which is a virtual function call), just use `bool maybeEmitMakeVectorFromScalar(IRType* type);` and `void maybeCloseMakeVectorFromScalar(bool v)`.